Firestore/Source/API/FIRFirestore.m

```diff
@@ -264,11 +276,17 @@ - (void)runTransactionWithBlock:(id _Nullable (^)(FIRTransaction *, NSError **er
 }
 
 - (void)shutdownWithCompletion:(nullable void (^)(NSError *_Nullable error))completion {
-  if (!self.client) {
+  FSTFirestoreClient *client;
+  @synchronized (self) {
+    client = _client;
+    _client = nil;
+  }
+
+  if (!client) {
     completion(nil);
```

Comment:
Uh, wow.  We probably shouldn't call their completion block synchronously from this thread.  I know this is existing code and we don't publicly expose shutdown, but maybe add a comment?

// TODO(mikelehen): We should be dispatching to the user queue, but since this isn't publicly exposed, we don't care for now.